lbry/schema/attrs.py

```diff
@@ -569,3 +666,305 @@ def append(self, tag: str):
+            if delete:
+                if k not in self:
+                    continue
+                elif len(v) > 0:
+                    self[k] -= v
+                else:
+                    del self[k]
+            else:
+                if k not in self:
+                    self[k] = v
+                else:
+                    self[k] |= v
+
+    def __getitem__(self, key):
+        if key in self.message.s:
+            return StringList(self.message.s[key])
+        raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        self.message.s[key].CopyFrom(value.message)
+
+    def __delitem__(self, key):
+        del self.message.s[key]
+
+    def __iter__(self):
+        return iter(self.message.s)
+
+    def __len__(self):
+        return len(self.message.s)
+
+class StringList(Metadata, MutableSet, Iterable):
+    __slots__ = ()
+    item_class = str
+
+    def __contains__(self, x):
+        return x in self.message.vs
+
+    def __iter__(self):
+        return iter(self.message.vs)
+
+    def __len__(self):
+        return len(self.message.vs)
+
+    def add(self, x):
+        if x not in self:
+            self.message.vs.append(x)
+
+    def discard(self, x):
+        for i, v in enumerate(self.message.vs):
+            if v == x:
+                del self.message.vs[i]
+
+class GenericExt(Metadata, MutableMapping, Iterable):
+    __slots__ = Metadata.__slots__ + ( 'field_descriptor', )
+
+    def __init__(self, message):
+        super().__init__(message)
+        self.field_descriptor = {}
+        for f, v in message.ListFields():
+            self.field_descriptor[f.name] = f
+
+    def merge(self, other: 'GenericExt', delete: bool = False) -> 'GenericExt':
+        if delete:
+            for f in other.message.ListFields():
+                if other.message.HasField(f.name):
+                    self.message.ClearField(f.name)
+        else:
+            self.message.MergeFrom(other.message)
+        return self
+
+    def __getitem__(self, key):
+        f = self.field_descriptor[key]
+        if f.type == FieldDescriptor.TYPE_MESSAGE and not self.message.HasField(key):
+            raise KeyError(key)
+        v = getattr(self.message, key)
+        return GenericExt(v) if f.type == FieldDescriptor.TYPE_MESSAGE else v
+
+    def __setitem__(self, key, value):
+        f = self.field_descriptor[key]
+        if f.type == FieldDescriptor.TYPE_MESSAGE:
+            msg = getattr(self.message, key)
+            msg.CopyFrom(value.message)
+        else:
+            setattr(self.message, key, value)
+
+    def __delitem__(self, key):
+        self.message.ClearField(key)
+
+    def __iter__(self):
+        present = filter(lambda f: self.message.HasField(f.name), self.message.ListFields())
+        return iter(map(lambda f: f.name, present))
+
+    def __len__(self):
+        present = filter(lambda f: self.message.HasField(f.name), self.message.ListFields())
+        return len(present)
```

Comment:
First attempt at a generic extension wrapper to be used when extension type is dynamic, not known at compile time. Most likely broken, though I did see __getitem__() work. To be removed.